### PR TITLE
Add Metrics for TektonChains

### PR DESCRIPTION
**Pull request by khrm**
Added Metrics `chains_reconciled` which tells us what type of storage, signer and format is being used during the installation of the chain.
This can be use for telemetry
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Metrics `chains_reconciled` added which has {oci/pipelinerun/taskrun}_{storage/format/signer} information as metrics tags.
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind feature

File: pkg/reconciler/kubernetes/tektonchain/metrics.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2021 The Tekton Authors
```
Comment (PuneetPunamiya) [inline pkg/reconciler/kubernetes/tektonchain/metrics.go:2]:
```suggestion
Copyright 2023 The Tekton Authors 😝 
```

File: pkg/reconciler/kubernetes/tektonchain/metrics.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	version            tag.Key
+	taskrunFormat      tag.Key
+	taskrunStorage     tag.Key
+	taskrunSigner      tag.Key
+	pipelinerunFormat  tag.Key
+	pipelinerunStorage tag.Key
+	pipelinerunSigner  tag.Key
+	ociFormat          tag.Key
+	ociStorage         tag.Key
+	ociSigner          tag.Key
```
Comment (PuneetPunamiya) [inline pkg/reconciler/kubernetes/tektonchain/metrics.go:50]: Not sure, but do we also need to include remaining keys ?

**Review by jkandasa — COMMENTED** [review]
/lgtm

**Pull request merged**